elastic/integrations
# [Cloud Security] Update manifest format_version to 3.0.0 for cloud_security_posture package

## Proposed commit message

making necessary changes to update `cloud_security_posture` package format_version to 3.0.0 to opt in it for serverless:
- updated `type: object` field  mappings either to more suitable types or added `object_type` as it's required now
- ~added DLM data_retention of 180d for findings index to enable data retention in serverless as ILM is not supported in serverless~ due to the [issue](https://github.com/elastic/integrations/pull/8162#issuecomment-1759649096) will be handled separately as a part of https://github.com/elastic/security-team/issues/7642
- added `security` capability so the package shows up in serverless




## Checklist


## Author's Checklist



## How to test this PR locally


- Install `elastic-package` and run `elastic-package check` inside `cloud_security_posture` package to validate
- Follow the steps in https://docs.google.com/document/d/1VbPZp_JgLuMpKixSGrQrZejARAJYje5zjoS3vmENF-s/edit#heading=h.7ymomq669pb4 to run stack and test the package

## Related issues


- Fixes https://github.com/elastic/security-team/issues/7624

## Screenshots
